test/cli/install/migration/complex-workspace/package.json

```diff
@@ -5,7 +5,6 @@
     "bar": "https://github.com/oven-sh/bun/raw/f7e4eb83694aa007a492ef66c28ffbe6a2dae791/test/cli/install/bar-0.0.2.tgz",
     "bun-types": "file:bun-types",
     "hello": "file:hello-0.3.2.tgz",
-    "install-test": "bitbucket:dylan-conway/public-install-test",
```

Comment:
@dylan-conway Interesting finding while debugging the test failures. Bitbucket is the only one that still does an ssh key check even if the repo is public. Github and Gitlab both clone with e.g.

```
git clone ssh://git@gitlab.com/org/repo.git
```

if the repo is public, even if the system does not have a linked ssh key. Bitbucket does a key check regardless.

Reply:
I see what you mean, not sure what's causing this yet

Reply:
@dylan-conway Found it! The package-lock.json has the bitbucket git dep listed as:

```
git+ssh://git@bitbucket.org/dylan-conway/public-install-test.git#......
```

This causes git clone to fail because of the `.git`. It checks for an SSH key for some reason and prompts for the password. With a fresh package, this does not happen, because the `.git` extension is ommitted in the lockfile version when installing those scp-like items. Is this just an unavoidable thing due to bitbucket?

For this to happen, a user would need:

1. An old npm lockfile
2. A bitbucket dependency listed in that lockfile the way npm prints it

Bun install can't work around it by .e.g. removing `.git` for bitbucket, because then it would skip using ssh keys on private repos.

The only thing I can think here is to force the https try to fail and not simply prompt for credentials. When https fails, fall back to ssh. If ssh fails, it's a complete failure.

How to stop git from prompting username & password:
```
GIT_TERMINAL_PROMPT=0 export GIT_ASKPASS=echo SSH_ASKPASS=echo GCM_INTERACTIVE=never git clone ...
```

But this does not seem like the right approach imo and think the bitbucket thing should just be a 'caveat'